aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -81,13 +76,13 @@ public abstract class S3V4RestSignerClient
   private static final String SCOPE = "sign";
 
   @SuppressWarnings("immutables:incompat")
-  private static volatile ScheduledExecutorService tokenRefreshExecutor;
+  private volatile AuthManager authManager;
 
   @SuppressWarnings("immutables:incompat")
-  private static volatile RESTClient httpClient;
+  private volatile AuthSession authSession;
 
   @SuppressWarnings("immutables:incompat")
-  private static volatile Cache<String, AuthSession> authSessionCache;
+  private volatile RESTClient httpClient;
```

Comment:
I'm a little concerned that this is going to be an issue.  It looks like we're making a separate `AuthMangaer`, `AuthSession`, and `RESTClient` for each signer instance.  I feel like we could reuse at least the http client by leaving it static (that potentially consumes the most resources).

Reply:
Are you sure? I always found this dangerous: 

https://github.com/apache/iceberg/blob/be9808fb3d75b1697fc4ed45c02602de9cfb6371/aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java#L192-L193

IOW the `HTTPClient` is static, but is built based on instance methods. If the value returned by `properties()` or by `baseSignerUri()` changes, the changes won't be reflected in a previously-created `HTTPClient` instance.

Reply:
Yes, this isn't something we can consider a lightweight resource and because we bind the FileIO to a table, this could result in significant numbers of instances being created especially in cases where we distribute these references.

It isn't ideal that there is potential for different property values, but I would take that risk over the impact of having runaway client creation.  The properties are mostly to configure aspects of the http implementation (I think only max retries is the only referenced value).

Reply:
OK, I will make `httpClient` static again 👍